<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="laser_frame_id" default="laser_link" />
    <arg name="video_device" default="/dev/video2" />
    <arg name="camera_frame_id" default="camera_link" />
    <arg name="image_view" default="false" />
    <arg name="use_rviz" default="false" />
    <arg name="use_proc" default="false" />

    <!-- USB camera -->
    <include file="$(find usb_cam)/launch/usb_cam.launch" >
        <arg name="video_device" value="$(arg video_device)" />
        <arg name="camera_frame_id" value="$(arg camera_frame_id)" />
        <arg name="image_view" value="$(arg image_view)" />
    </include>

    <!-- Hokuyo laserscan -->
    <include file="$(find hokuyo_node)/launch/laserscan.launch" >
        <arg name="laser_frame_id" value="$(arg laser_frame_id)" />
    </include>


    <!-- Static transformation -->
    <!-- <node name="base2camera" pkg="tf" type="static_transform_publisher" 
            args="0 0 0 0 0 0 1 
                /base_link /camera_link 20"/>
    <node name="base2laser" pkg="tf" type="static_transform_publisher" 
            args="0 0 0.1 0 0 0 1 
                /base_link /laser_link 20"/> -->


    <!-- RVIZ -->
    <group if="$(eval arg('use_rviz') == true)">
        <node pkg="rviz" type="rviz" name="rviz_node" 
            args="-d $(find active_walker)/rviz/default_setting.rviz"/>
    </group>

    <!-- Image proc -->
    <group if="$(eval arg('use_proc') == true)">
        <remap from="image_raw" to="/usb_cam/image_raw" />
        <remap from="camera_info" to="/usb_cam/camera_info" />
        <node name="image_proc_node" pkg="image_proc" type="image_proc" />
    </group>


</launch>